sponsorship_compassion/models/contracts.py

```diff
@@ -249,7 +249,38 @@ def create(self, vals):
                 'sponsor_id': vals.get('correspondent_id', vals['partner_id'])
             })
 
-        return super(SponsorshipContract, self).create(vals)
+        """ Generates commitment number for contracts BVRs
+        """
+        if 'commitment_number' not in vals:
+            partner_id = vals.get('partner_id')
+            correspondent_id = vals.get('correspondent_id')
+            if partner_id:
+                other_nums = self.search([
+                    ('partner_id', '=', partner_id)
+                ]).mapped('commitment_number')
+
+                other_nums.extend(self.search([
+                    ('partner_id', '=', correspondent_id)
+                ]).mapped('commitment_number'))
```

Comment:
You could have only one search if you put the following criterias:
```
'|', ('partner_id', '=', partner_id),
('partner_id', '=', correspondent_id)
```

Reply:
Maybe you should as well count the contracts of the correspondent ? It would become : 
```
'|', '|', '|', ('partner_id', '=', partner_id),
('partner_id', '=', correspondent_id),
('correspondent_id', '=', partner_id),
('correspondent_id', '=', correspondent_id),
```